dataframe-jdbc/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/readJdbc.kt

```diff
@@ -437,6 +551,38 @@ public fun DataFrame.Companion.getSchemaForSqlQuery(connection: Connection, sqlQ
     }
 }
 
+/**
+ * Retrieves the schema of an SQL query result or the SQL table using the provided database configuration.
+ *
+ * @param [sqlQueryOrTableName] the SQL query to execute and retrieve the schema from.
+ * @return the schema of the SQL query as a [DataFrameSchema] object.
+ */
+public fun DatabaseConfiguration.getDataFrameSchema(
```

Comment:
maybe `getDataFrameSchemaFor(sqlQueryOrTableName)`, otherwise a good name :)

Reply:
better be unified with other schema names (and rename them together)